pkg/kubelet/kubelet.go

```diff
@@ -3286,6 +3286,15 @@ func (kl *Kubelet) generateAPIPodStatus(pod *api.Pod, podStatus *kubecontainer.P
 	s.Phase = GetPhase(spec, s.ContainerStatuses)
 	kl.probeManager.UpdatePodStatus(pod.UID, s)
 	s.Conditions = append(s.Conditions, status.GeneratePodReadyCondition(spec, s.ContainerStatuses, s.Phase))
+	// We want to keep PodScheduled condition. If for any reason it's not set we should do it now as the pod
```

Comment:
It took me a while to figure out this logic. I would rewrite the comment as follows:

"s (the PodStatus we are creating) will not have a PodScheduled condition yet, because converStatusToAPIStatus() does not create one. If the existing PodStatus has a PodScheduled condition, then copy it into s and make sure it is set to true. If the existing PodStatus does not have a PodScheduled condition, then create one that is set to true."